libraries/stdlib/js/src/kotlin/text/regex.kt

```diff
@@ -84,18 +84,19 @@ public actual class Regex public actual constructor(pattern: String, options: Se
     private var nativeMatchesEntirePattern: RegExp? = null
     private fun initMatchesEntirePattern(): RegExp =
         nativeMatchesEntirePattern ?: run {
-            if (pattern.startsWith('^') && pattern.endsWith('$'))
-                nativePattern
-            else
-                return RegExp("^${pattern.trimStart('^').trimEnd('$')}$", options.toFlags("gu"))
-        }.also { nativeMatchesEntirePattern = it }
-
+            val delazifiedPattern = pattern
+                .replace("*?", "*")
```

Comment:
If the pattern was `aa\\*?bb`, delazified one will become `aa\\*bb`. While the original pattern accepts both `aabb` and `aa*bb`, the delazified one accepts only `aa*bb`.

Reply:
Aaargh, damn regex-manipulation, always a hassle.
Without getting the delazifying fixed up, I do not see a reliable way to do `matches` or `matchEntire` for JS with `MULTILINE` but only see as option to throw a not-supported exception in that situation.

Reply:
Ok, next try.
If I made such a stupid error again, I should really get some sleep. :-D